ingestion/portland_ingest.py

```diff
@@ -21,131 +21,164 @@
+            df["season"] = "Summer"
+        elif "fall" in value:
+            df["season"] = "Fall"
+        elif "winter" in value:
+            df["season"] = "Winter"
+
+        # get day_type from filename
+        if "sat" in value:
+            df["day_type"] = "Saturday"
+        elif "weekday" in value:
+            df["day_type"] = "Weekday"
+        elif "sunday" in value:
+            df["day_type"] = "Sunday"
+        data_frames[key] = df
+    return data_frames
+
+
+def format_data(all_rider_data):
+    """
+    standarize column formats and keep the relevant
+    data fields needed to match our data model, or
+    merge the data at the datetime level
+    """
+    for key, df in all_rider_data.items():
+        df.columns = df.columns.str.lower().str.replace(" ", "_")
+        all_rider_data[key] = df[["location_id", "ons", "day_type", "season"]]
+    return all_rider_data
 
-# flag seasons, all days are flagged to Spring, except for summer months,
-# to differenciate for routes that only run during the summer months
-calendar_df["season"] = "Spring"
 
-# Summer 2023 date range
-summer_start = pd.to_datetime("2023-06-21")
-summer_end = pd.to_datetime("2023-09-22")
+def create_calendar_df(start_date, end_date, season_ranges):
+    """
+    Create a dataframe with datetime values as well as
+    the corresponding season and day type associated with
+    each datetime, return a dataframe
+    """
+    calendar_dates = pd.date_range(start=start_date, end=end_date, freq="D")
+    calendar_df = pd.DataFrame(calendar_dates, columns=["date"])
+    calendar_df["day_type"] = calendar_df["date"].dt.day_name()
+    calendar_df["day_type"] = (
+        calendar_df["day_type"]
+        .map({"Saturday": "Saturday", "Sunday": "Sunday"})
+        .fillna("Weekday")
+    )
+    calendar_df["season"] = "Spring"
+
+    # label seasons based on date ranges
+    for season, (season_start, season_end) in season_ranges.items():
+        calendar_df.loc[
+            (calendar_df["date"] >= season_start) & (calendar_df["date"] < season_end),
+            "season",
+        ] = season
+
+    return calendar_df
+
+
+def merge_data(all_rider_data, calendar_df):
+    """
+    Combines two dataframes by day type and season value,
+    returns a dataframe with the columns from both data frames
+    """
+    merged_data = pd.merge(
+        all_rider_data, calendar_df, on=["day_type", "season"], how="left"
+    )
+    return merged_data.sort_values(by=["date", "day_type", "season"], ascending=True)
 
-calendar_df.loc[
-    (calendar_df["date"] >= summer_start) & (calendar_df["date"] <= summer_end),
-    "season",
-] = "Summer"
 
-# combine Tri Met data with calendar dates in order to convert the data set
-# from day_type (sat, sun, or weekday) + season level into date level
+def preprocess_location_id(location_id):
+    """
+    Add a city prefix to make the location id more specific
+    for our data model
+    """
+    return f"Portland-{location_id}"
```

Comment:
I think for now we will handle city as one column and location id as another from the db perspective so no need to combine those for ingestion, but fine to leave for now until db stage.